### PR TITLE
Support passing max item width to grid view

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
- Add `maxItemWidth` prop to GridView to support a responsive item width without having to know number of colums
- Minor refactor - keep itemSize on state instead on class since it affect the render cycle

WOAUILIB-2207

## Changelog
GridView now supports `maxItemWidth` prop for a responsive item size (this is instead of passing `numColumns` prop)

### CODE REVIEW

File: generatedTypes/src/components/gridView/index.d.ts [hunk 1]
```diff
@@ -29,28 +33,32 @@ export interface GridViewProps {
...
     viewWidth: number;
     numColumns: number;
+    itemSize: number;
 }
-declare type ExistProps = GridViewProps & State;
 /**
  * @description: A auto-generated grid view that calculate item size according to given props
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/GridViewScreen.tsx
  */
-declare class GridView extends UIComponent<GridViewProps, State> {
```
Comment (Inbal-Tish) [inline generatedTypes/src/components/gridView/index.d.ts:44]: Why you're not passing the 'State'?

File: src/components/gridView/index.tsx [hunk 2]
```diff
@@ -38,44 +43,38 @@ export interface GridViewProps {
...
+  itemSize: number;
 }
 
-
-type ExistProps = GridViewProps & State
 /**
  * @description: A auto-generated grid view that calculate item size according to given props
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/GridViewScreen.tsx
  */
-class GridView extends UIComponent<GridViewProps, State> {
```
Comment (Inbal-Tish) [inline src/components/gridView/index.tsx:56]: Why you're not passing 'State'?

File: src/components/gridView/index.tsx [hunk 3]
```diff
@@ -38,44 +43,38 @@ export interface GridViewProps {
...
+class GridView extends UIComponent<GridViewProps> {
   static displayName = 'GridView';
 
   static defaultProps = {
     numColumns: 3,
```
Comment (Inbal-Tish) [inline src/components/gridView/index.tsx:65]: DEFAULT_NUM_COLUMNS instead of "3"

File: src/components/gridView/index.tsx [hunk 4]
```diff
@@ -38,44 +43,38 @@ export interface GridViewProps {
...
     itemSpacing: Spacings.s4
```
Comment (Inbal-Tish) [inline src/components/gridView/index.tsx:66]: Better take this too to const as you're using it again as default in line #136

File: src/components/gridView/index.tsx [hunk 5]
```diff
@@ -38,44 +43,38 @@ export interface GridViewProps {
   /**
    * whether to keep the items initial size when orientation changes,
    * in which case the apt number of columns will be calculated automatically.
+   * Irrelevant when passing maxItemWidth
```
Comment (Inbal-Tish) [inline src/components/gridView/index.tsx:46]:
```suggestion
   * Ignored when passing 'maxItemWidth'
```

File: src/components/gridView/index.tsx [hunk 6]
```diff
@@ -1,15 +1,16 @@
 import _ from 'lodash';
 import React from 'react';
 import {StyleSheet} from 'react-native';
-// TODO: we should use asBaseComponent here instead of using UIComponent directly
 import {Colors, Spacings} from 'style';
+// TODO: we should use asBaseComponent here instead of using UIComponent directly
```
Comment (Inbal-Tish) [inline src/components/gridView/index.tsx:5]: Don't you want to include this in your refactor?

Comment (ethanshar) [reply src/components/gridView/index.tsx:5]: not at the moment..

File: src/components/gridView/index.tsx (hunk 2 above)
Comment (ethanshar) [reply src/components/gridView/index.tsx:56]: Fixed

File: src/components/gridView/index.tsx (hunk 3 above)
Comment (ethanshar) [reply src/components/gridView/index.tsx]: Fixed

File: src/components/gridView/index.tsx (hunk 4 above)
Comment (ethanshar) [reply src/components/gridView/index.tsx]: Done

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**